### ISSUE TITLE
BlendTree(Direct)やAdditiveレイヤーのStateでWriteDefaultsをオンに保つオプションをMAMergeAnimatorに用意する

### ISSUE THREAD

**Issue by UnisakiStudio** [bdunderscore/modular-avatar#771]
MA Merge Animatorには「アバターのWrite Defaults設定に合わせる」という設定があるが、
これを使うと（当たり前ではあるが）すべてのStateのWrite Defaultsが変更される。
しかし以下のリンク先情報のように、DirectのBlendTreeやAdditiveLayerのStateにおいてWriteDefaultsがオフだと不具合が発生することがある。
そのため、DirectのBlendTreeやAdditiveLayerのStateについてはWriteDefaultsがオンになる（あるいはオンのままになる）ことを選択できることが望ましいと思われる。
なお、VRCFuryのFix WriteDefaultsコンポーネントを使用すると上書きされるため一応解決される。

https://vrcfury.com/technical/wd/#direct-blendtrees-and-additive-layers-with-wd-off

**Comment by bdunderscore** [comment]
確かに、WD調整から除外するべきですが、互換性にかかわる変更（既存ギミックに影響を与えかねない）のため要注意ですね・・・移行方法については少し考えます

**Comment by Narazaka** [comment]
同じ問題についてのissue立てちゃった……
https://github.com/bdunderscore/modular-avatar/issues/891

基本的に同様で、ギミック作者とかもDirect BlendTreeとかを使おうとすると普通にハマるっぽいのでどうにかしたい
https://x.com/narazaka/status/1804200866964291757

**Comment by KisaragiEffective** [comment]
現行は
> ### MA Merge Animator
> - [ ] WDをアバターに合わせる

となっていますが、

> ### MA Merge Animator
> - [ ] WDをアバターに合わせる
>   - [ ] DirectBlendTreeは変換しない
>   - [ ] Additiveレイヤーのステートは変換しない

のようにオプトインにしてほしさがあります。こうすることで影響を与えることがないはずです。

Additiveレイヤーはともかく、DirectBlendTreeはMaterialのfloatプロパティを線形補間したい時にそれなりに使います。
ただ、これを真冬ちゃんなどWDがオフで組まれているアバターにMergeするとDBTまでWDがオフになってしまい、少なくともエディタ上ではうまく動きませんでした。
